src/tickit/core/management/schedulers/master.py

```diff
@@ -131,3 +135,17 @@ def sleep_time(self, when: SimTime) -> float:
             (when - self.ticker.time) / self.simulation_speed
             - (time_ns() - self.last_time)
         ) / 1e9
+
+    async def handle_component_exception(self, message: ComponentException) -> None:
+        """Handle exceptions raised from componenets by shutting down the simulation.
+
+        If a component produces an exception, the scheduler will produce a message to
+        all components in the simulation to cause them to cancel any running component
+        tasks. After which the scheduler shuts its self down.
+
+        """
+        await super().handle_component_exception(message)
+        # Might need to check order of 'self.ticker.finished.set()' and
+        # ' self.error.set()' when running with an error
+        # that is not in "do inital tick"
```

Comment:
Let's figure this one out (and back it up with some tests) before merging :sweat_smile: